samples/Twitter.gs

```diff
@@ -0,0 +1,86 @@
+var CLIENT_ID = '...';
+var CLIENT_SECRET = '...';
+
+/**
+ * Authorizes and makes a request to the Twitter API v2
+ * OAuth 2.0 Making requests on behalf of users
+ * https://developer.twitter.com/en/docs/authentication/oauth-2-0/user-access-token
+ */
+function run() {
+  var service = getService();
+  if (service.hasAccess()) {
+    // https://developer.twitter.com/en/docs/twitter-api/users/lookup/api-reference/get-users-by-username-username
+    var url = `https://api.twitter.com/2/users/by/username/workspacedevs?user.fields=verified`;
+    var response = UrlFetchApp.fetch(url, {
+      headers: {
+        Authorization: 'Bearer ' + service.getAccessToken()
+      },
+      muteHttpExceptions: true
+    });
+    var result = JSON.parse(response.getContentText());
+    Logger.log(JSON.stringify(result, null, 2));
+  } else {
+    var authorizationUrl = service.getAuthorizationUrl();
+    Logger.log('Open the following URL and re-run the script: %s',
+      authorizationUrl);
+  }
+}
+
+/**
+ * Reset the authorization state, so that it can be re-tested.
+ */
+function reset() {
+  getService().reset();
+}
+
+/**
+ * Configures the service.
+ */
+function getService() {
+  return OAuth2.createService('Twitter')
+    // Set the endpoint URLs.
+    .setAuthorizationBaseUrl(
```

Comment:
If supplying PKCE parameters, would be better to show how to do it properly.

* the code_challenge_method should be S256. Plaintext is much weaker.
* the code_challenge value should be a the base64 encoded sha256 hash of a randomly generated value
* The raw challenge value needs to be durable so it can be provided in the auth callback

Granted, PKCE is meant more for public clients that can't keep secrets, so less of an issue for apps script. Also we may want to see if we can just build PKCE support into the library itself since it's standardized.

Reply:
@sqrrrl — was able to accommodate the following changes:

- changed `code_challenge_method` from `plain` to `S256`
- the `code_challenge` value is now a base64 encoded sha256 hash of a randomly generated value